src/python/turicreate/test/test_io_s3.py

```diff
@@ -129,6 +129,20 @@ def test_s3_sframe_upload(self, folder):
         # we can trust the upload becuase if the upload fails,
         # s3 will respond with 5xx
 
+    @pytest.mark.parametrize(
+        "url",
+        ["s3://gui/dummy", "./willy-nily/blah", "https://foo.com", "http://hao.com"],
```

Comment:
Probably just one invalid URL is enough. What the benefit in testing this more than once?

Reply:
The difference is in the protocol: s3, local relative, https, http. They all go to different code paths.